components/rating/rating.tsx

```diff
@@ -0,0 +1,124 @@
+import { Icon, Props as IconProps } from '@geist-ui/react-icons'
+import React, { useEffect, useMemo, useState } from 'react'
+
+import { GeistUIThemes } from '../themes/presets'
+import { NormalTypes } from '../utils/prop-types'
+import Star from '@geist-ui/react-icons/star'
+import useTheme from '../use-theme'
+import withDefaults from '../utils/with-defaults'
+
+interface Props {
+  type?: NormalTypes
+  className?: string
+  icon?: (props: IconProps) => React.ReactElement<Icon>
+  count?: 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10
+  value?: 0 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9
+  valueCallback?: React.Dispatch<React.SetStateAction<number>>
```

Comment:
Maybe it would be better to change its name to `onValueChange` or `onChange`? (We try to be as consistent as possible with the events of other components)

Reply:
Migrating from type `React.SetStateAction` to a normal function type: `(value: number) => void`.

Reply:
There may be some changes here, and you will also need to change some test cases. My idea is to keep the public APIs as simple as possible.

Reply:
```suggestion
  onValueChange?: (value?: any) => void
```

I am no expert on the type consequences for this change, but I guess moving to a normal function type decouples the callback/state change from the internal React state management? So your proposal shouldn't break anything, instead just allow users to pass more functionality than Reacts dispatched state setter? If I understand this right, then a normal function type is the better choice.

Reply:
Yes, the component should be kept internally and externally isolated, and even if a function is added to the props, this function should be transparent. If you have questions or have difficulty solving them, you can leave them alone and I will test and optimize them locally when you are done with them subsequently.